mantidproject/mantid
# Avoid SIGABRT in CompAssembly::getComponentByName

`CompAssembly::getComponentByName` takes a string holding the component name as input. In case no component of that name exist, an exception should be thrown rather than to stop Mantid executing (sigabrt).

**To test:**

Choose a file (likely a loader located in `DataHandling`), which uses `CompAssembly::getComponentByName` and its corresponding instrument definition file, then try loading.

For example:
- choose `LoadILLReflectrometry` and replace the component name  `uniq_detector` of its instrument definition file `D17_Definition.xml` with another name
- execute `LoadILLReflectometry`, it should result in an error but should not stop Mantid executing


Fixes #19002.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
